### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by brian-weir**

### CODE REVIEW

File: triggers/daml/AutoApproval.daml [hunk 1]
```diff
@@ -167,7 +167,11 @@ generateCommands choiceArg queryResults = map (\(cid, _) -> (toAnyContractId cid
 generateCommandsWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (t2 -> t -> Bool) -> (ContractId t2 -> c) -> [(ContractId t2, t2)] -> [(ContractId t, t)] -> [(AnyContractId, Command)]
 generateCommandsWithCid _ _ [] _ = []
 generateCommandsWithCid _ _ _ [] = []
-generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> (\(cid, _) -> (toAnyContractId cid, exerciseCmd cid $ toChoice reqCid)) <$> findFirstMatch req targets pred) requests
+generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> getCommandWithCid toChoice reqCid <$> findFirstMatch req targets pred) requests
+
+-- | Takes a choice and a contract id which is a parameter of that choice, the contract which exposes this choice and builds up a command
+getCommandWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (ContractId t2 -> c) -> ContractId t2 -> (ContractId t, t) -> (AnyContractId, Command)
+getCommandWithCid toChoice requestCid (cid, _) = (toAnyContractId cid, exerciseCmd cid $ toChoice requestCid)
```
Comment (brian-weir) [inline triggers/daml/AutoApproval.daml:174]: The goal of doing this is to make it easier to comprehend what this function is doing, by removing one of the lambda's out into its own function.

File: triggers/daml/AutoApproval.daml [hunk 2]
```diff
@@ -167,7 +167,11 @@ generateCommands choiceArg queryResults = map (\(cid, _) -> (toAnyContractId cid
 generateCommandsWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (t2 -> t -> Bool) -> (ContractId t2 -> c) -> [(ContractId t2, t2)] -> [(ContractId t, t)] -> [(AnyContractId, Command)]
 generateCommandsWithCid _ _ [] _ = []
 generateCommandsWithCid _ _ _ [] = []
-generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> (\(cid, _) -> (toAnyContractId cid, exerciseCmd cid $ toChoice reqCid)) <$> findFirstMatch req targets pred) requests
+generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> getCommandWithCid toChoice reqCid <$> findFirstMatch req targets pred) requests
+
+-- | Takes a choice and a contract id which is a parameter of that choice, the contract which exposes this choice and builds up a command
+getCommandWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (ContractId t2 -> c) -> ContractId t2 -> (ContractId t, t) -> (AnyContractId, Command)
```
Comment (lucianojoublanc-da) [inline triggers/daml/AutoApproval.daml:173]: Do you need all these type constraints on here? e.g. `Choice` constraint? I recall we removed it from a similar function.

**Review by lucianojoublanc-da — APPROVED** [review]

File: triggers/daml/AutoApproval.daml (hunk 2 above)
Comment (brian-weir) [reply triggers/daml/AutoApproval.daml:173]: Only when I remove the entire type signature for the function, it works. 

Removing the `choice` constraint causes `Could not deduce (HasExercise t c r0) arising from a use of ‘exerciseCmd’` error. Also, removing the type constraint while leaving the rest of the signature causes an error.

File: triggers/daml/AutoApproval.daml (hunk 1 above)
Comment (lucianojoublanc-da) [reply triggers/daml/AutoApproval.daml:174]: What I mean is, can you just remove `forall t c r t2. (Template t, Choice t c r, Template t2) =>`, or parts of it? Do you really need that type constraint?

Comment (brian-weir) [reply triggers/daml/AutoApproval.daml:174]: The best I can reduce it the type constraint to without an error is `forall t c r. Choice t c r =>`.

Comment (brian-weir) [reply triggers/daml/AutoApproval.daml]: I've updated this signature and a few others that could also be simplified.

**Pull request merged**